skyportal/handlers/api/photometry.py

```diff
@@ -872,12 +877,112 @@ def add_external_photometry(json, user, parent_session=None):
             )
         )
-        ids, upload_id = insert_new_photometry_data(
-            df, instrument_cache, group_ids, stream_ids, user, session
-        )
-        log(
-            f'Request from {username} with {len(df.index)} rows complete with upload_id {upload_id}'
-        )
+        if duplicates in ["ignore", "update"]:
+            values_table, condition = get_values_table_and_condition(df)
+
+            new_photometry_query = session.execute(
+                sa.select(values_table.c.pdidx)
+                .outerjoin(Photometry, condition)
+                .filter(Photometry.id.is_(None))
+            )
+
+            new_photometry_df_idxs = [g[0] for g in new_photometry_query]
+
+            duplicated_photometry = (
+                session.execute(
+                    sa.select(values_table.c.pdidx, Photometry)
+                    .join(Photometry, condition)
+                    .options(joinedload(Photometry.groups))
+                    .options(joinedload(Photometry.streams))
+                )
+                .unique()
+                .all()
+            )
+
+            id_map = {}
+
+            for df_index, duplicate in duplicated_photometry:
+                id_map[df_index] = duplicate.id
+
+                if duplicates in ["ignore"]:
+                    continue
+
+                duplicate_group_ids = {g.id for g in duplicate.groups}
+                duplicate_stream_ids = {s.id for s in duplicate.streams}
+
+                # posting to new groups?
+                if len(set(group_ids) - duplicate_group_ids) > 0:
+                    # select old + new groups
+                    group_ids_update = set(group_ids).union(duplicate_group_ids)
+                    groups = (
+                        session.execute(
+                            sa.select(Group).filter(Group.id.in_(group_ids_update))
+                        )
+                        .scalars()
+                        .all()
+                    )
+                    # update the corresponding photometry entry in the db
+                    duplicate.groups = groups
+                    log(
+                        f'Adding groups {group_ids_update} to photometry {duplicate.id}'
+                    )
+
+                # posting to new streams?
+                if stream_ids:
+                    # Add new stream_photometry rows if not already present
+                    stream_ids_update = set(stream_ids) - duplicate_stream_ids
+                    if len(stream_ids_update) > 0:
+                        for id in stream_ids_update:
+                            session.add(
+                                StreamPhotometry(
+                                    photometr_id=duplicate.id, stream_id=id
+                                )
+                            )
+                        log(
+                            f'Adding streams {stream_ids_update} to photometry {duplicate.id}'
+                        )
+
```

Comment:
Should we also log here at the end that photometry did not need updating?

Reply:
Might end up with a **lot** of logs that way. I think it's fine without it. Any specific reason why you'd like to see that info?
I think what could be ok is keeping track of which were not updated in a list, and then posting just one single log with that list.

Reply:
Ok that seems good. Just want to simplify debugging if needed!

Reply:
Agreed! I'll add that one line log thing now.

Reply:
Just added something. Let me know if that's what you expected.

Reply:
Yes ok!